### DISCUSSION TITLE
Some confusions on recipes

### DISCUSSION THREAD

**Discussion by Leedehai** [tauri-apps/tauri#4010] [Q&A] [answered]
Hello Tauri experts, while reading the docs ([e4ec76](https://github.com/tauri-apps/tauri-docs/tree/e4ec760fce919f6abaa08e50e8b12238ded1bce7)) I got some questions:

**Q1**
Is it correct that `tauri.conf.json` support comments? Could you make it clear in the doc?

**Q2**
The `tauri.conf.json` configs for Hermit[^hermit-recipe] and Cloudish[^cloudish-recipe] seem to be the same - that is, only one entry `"all": false` in `allowlist`, though the comments are different.

Is that right? If so, could you explain the doc why?

**Q3**
It seems there isn't a doc on what those config keys in the `allowlist` in file `tauri.conf.json` can be and what they are used for. Is there?

Especially, in the Bridge recipe[^bridge-recipe] there is already an `"all": false`, but then after it a long list of other things are listed and set to `false` as well. So it seems a mere `"all": false` is not enough to turn the other things off? If so, could you explain why? If not, could you explain what's the difference between this recipe and Hermit[^hermit-recipe] and Cloudish[^cloudish-recipe], which also say `"all": false`?

**Q4**
An app with a splash screen needs multiple windows, but its doc[^splash-screen] doesn't mention which recipe to use. However, the doc on the Multiwin recipe[^multiwin-recipe] seems to imply to have multiple windows one needs to use this recipe.

Am I right? If so, could you add that to the doc on splash screens?

Thanks!

[^hermit-recipe]: https://tauri.studio/docs/architecture/recipes/hermit
[^cloudish-recipe]: https://tauri.studio/docs/architecture/recipes/cloudish
[^bridge-recipe]: https://tauri.studio/docs/architecture/recipes/bridge
[^splash-screen]: https://tauri.studio/docs/guides/splashscreen
[^multiwin-recipe]: https://tauri.studio/docs/architecture/recipes/multiwin

**Comment by FabianLars** [comment, marked as answer]
- 1: Yes if it's JSON5, we will mention this in our rewritten docs, thx.
- 2-4: Don't focus on the recipes, there are still from alpha versions and don't apply that well after all the changes (especially after the removal of the built-in localhost server). They will either be updated before v1 or straight up removed. The general recommended approach in this regard is to start with a disabled allowlist and allow only the endpoint you need when you need them.

**Reply by Leedehai** [reply]
Thanks! Still, could you enumerate (like, when rewriting the docs) what the entries in the allowlist can be, as well as the role of `"all": false`?
